dask/array/core.py

```diff
@@ -1502,6 +1502,10 @@ def name(self, val):
             "please set ``._name``"
         )
 
+    def __iter__(self):
+        for i in range(len(self)):
+            yield self[i]
```

Comment:
I wonder if it would make sense to capture this in an HLG. WDYT?

Reply:
@jakirkham this would be done by #7655 I believe. I think it makes more sense to address as the slicing level. Making simple slices into HLGs might help with many other cases like this (#7832 for example).

Reply:
Wouldn't this still create a bunch of nodes for each slice we would do?

Reply:
Yes, but I think there's no way around that. The point of `__iter__` is to create a bunch of separate objects that user code can then do whatever it wants to. You could apply a different operation to every single slice, for example.

If you know you want to map a function over each element in axis 0, then you'd need to use some other interface (`map_blocks`) anyway.

Though someday, I'd love to see an HLG optimization pass that could rewrite
```python
da.stack([func(x) for x in arr])
```
into a blockwise operation (by detecting within the concatenation that all the inputs were Blockwise and had equivalent subgraphs).

Reply:
> Though someday, I'd love to see an HLG optimization pass that could rewrite
> 
> ```python
> da.stack([func(x) for x in arr])
> ```
> 
> into a blockwise operation (by detecting within the concatenation that all the inputs were Blockwise and had equivalent subgraphs).

I'd really like to see this too